### PR TITLE
feat(aws): Add new KMS check to prevent unintentional key deletion

### PR DESCRIPTION
**Pull request by danibarranqueroo**
### Context

To improve data security and ensure adherence to best practices, I'm introducing a new check within the KMS framework. This check ensures that AWS KMS keys are not deleted unintentionally.

A KMS key, before being deleted, is scheduled for deletion. This means that if you mistakenly delete a key, you’ll have time to recover it because it isn’t deleted immediately. When a key is scheduled for deletion, there is a mandatory period (30 days by default) before it is permanently deleted. 

### Description

I have added this new check called `kms_cmk_not_deleted_unintentionally` with his respective test to prove that it works.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.metadata.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "Data Deletion Protection"
+  ],
+  "ServiceName": "kms",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:partition:kms:region:account-id:certificate/resource-id",
+  "Severity": "critical",
+  "ResourceType": "AwsKmsKey",
+  "Description": "Ensure there is no customer keys scheduled for deletion.",
+  "Risk": "KMS keys cannot be recovered once deleted, also, all the data under a KMS key is also permanently unrecoverable if the KMS key is deleted.",
+  "RelatedUrl": "https://docs.aws.amazon.com/securityhub/latest/userguide/kms-controls.html#kms-3",
```
Comment (puchy22) [inline prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.metadata.json]: Could you put this link in Remediation.Other and here a link to KMS docs.

**Review by puchy22 — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.py
Comment (MrCloudSec) [inline prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.py:1]: Could you add another status when the key is being used, please?

File: prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                )
+                report.resource_id = key.id
+                report.resource_arn = key.arn
+                if key.state == "PendingDeletion":
+                    report.status = "FAIL"
+                    report.status_extended = (
+                        f"KMS CMK {key.id} is scheduled for deletion."
+                    )
+                elif key.state == "Enabled":
+                    report.status_extended = f"KMS CMK {key.id} is being used."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.py]: I meant by putting at the beginning of the loop:
`if key.state == "Enabled" or kms_client.provider.scan_unused_services`

Comment (MrCloudSec) [reply prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.py]: And adding a test for the case of an unused key

Comment (danibarranqueroo) [reply prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.py]: Done here https://github.com/prowler-cloud/prowler/pull/4595/commits/a1316eaa33b1ded3062bdf30afc7d49886cd3c17

File: prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.metadata.json (hunk 1 above)
Comment (danibarranqueroo) [reply prowler/providers/aws/services/kms/kms_cmk_not_deleted_unintentionally/kms_cmk_not_deleted_unintentionally.metadata.json]: Done here https://github.com/prowler-cloud/prowler/pull/4595/commits/de5b08518e601e92cda14ee3ff9fd932d502dbaf

**Review by MrCloudSec — APPROVED** [review]
Great job @danibarranqueroo !! 👏🏼

**Review by puchy22 — APPROVED** [review]
Thanks for the PR @danibarranqueroo! 🚀

**Pull request merged**